### DISCUSSION TITLE
[✨] Qwik-city Link preventdefault:click

### DISCUSSION THREAD

**Discussion by diicar** [QwikDev/qwik-evolution#278] [Proposals For Qwik]
### Is your feature request related to a problem?

Hello,

The preventdefault:click on <**Link href="/qwik"**>  is ignored.

### Describe the solution you'd like

Take into account preventdefault:click when using <**Link** /> component and do not trigger navigation when its present.

https://github.com/BuilderIO/qwik/blob/275e1c7723a0cf95d8e910514a8047a217bb137f/packages/qwik-city/runtime/src/link-component.tsx#L25

### Describe alternatives you've considered

None

### Additional context
Thank you

**Comment by manucorporat** [comment]
What's the use case for this?

**Comment by diicar** [comment]
Structure of the app:  
- routes : 
  - [...workspace]
      - index.tsx <-- acts as dashboard
      - item
        - [id] 
          - index.tsx <-- opens a modal with details    

On top of all features its provides I need it for 2 uses cases 

1. prevent scroll to top on click:
    - Imagine a page with a lot of items (with scroll)
      - Scroll a little bit
      - Click on one item
      - trigger the navigation (URL updated)
      - show a modal with details of that item
      - also scroll to top <- I want to prevent that.
2. replace state : 
    - Sometimes I need to be able to replaceState (history API)

**Comment by MattHeslington** [comment]
> What's the use case for this?

A like button on a card which is wrapped by Link tags - clicks to the Like button propagate through to the Link

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.

**Comment by wmertens** [comment]
Actually, [Link already handles preventdefault:click](https://github.com/QwikDev/qwik/blob/05d9c8517efc11034dc1b2d0546969f94e3adb65/packages/qwik-city/src/runtime/src/link-component.tsx#L126-L128), either when you specify it yourself, or when the link is a navpath.

The issue here is actually that aforementioned button is inside the link and that should get the `preventdefault:click` on it.
